### PR TITLE
CI improvements and cleanup for Cypress tests

### PR DESCRIPTION
**Pull request by U-DON**
## Description
- Run Cypress and Nightwatch tests in parallel in Jenkins.
    - Removed the Puppeteer tests and the related Slack notification.
    - Builds take about 10 minutes less time now.
- Added CI flag to Cypress so tests can be skipped in CI environments.
- Extended timeout on build step for Circle.
    - This is in response to recent builds occasionally running over 10 minutes without output.
- Removed unnecessary extended timeout for Cypress tests in Circle.
- Fixed some form tester documentation.

## Testing done
Tests pass in CI. Verified that skipping tests based on the `'CI'` environment variable works.

## Acceptance criteria
- [ ] E2E tests should continue to pass.
- [ ] Cypress tests should have access to the `'CI'` environment variable.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -187,6 +187,7 @@ jobs:
       - run:
           name: Build apps
           command: yarn build:webpack --env.buildtype vagovprod --env.scaffold
+          no_output_timeout: 15m
```
Comment (billfienberg) [inline .circleci/config.yml:190]: Docs: https://circleci.com/docs/2.0/configuration-reference/#run

Comment (billfienberg) [inline .circleci/config.yml:190]: Curious: What was the motivation reducing the time from `20m` to `15m`?

File: src/platform/testing/e2e/cypress/support/form-tester/README.md [hunk 2]
```diff
@@ -542,11 +550,6 @@ const testConfig = createTestConfig(
 
         // Fill out the rest of the page like normal.
         cy.fillPage();
-
-        // Don't forget to click continue!
```
Comment (billfienberg) [inline src/platform/testing/e2e/cypress/support/form-tester/README.md:546]: Is the **Continue** click no longer necessary?

**Review by billfienberg — APPROVED** [review]
I think this is safe to merge.

File: src/platform/testing/e2e/cypress/support/form-tester/README.md (hunk 2 above)
Comment (U-DON) [reply src/platform/testing/e2e/cypress/support/form-tester/README.md:546]: It's done automatically now. Just forgot to update the doc when that change was made.

File: .circleci/config.yml (hunk 1 above)
Comment (U-DON) [reply .circleci/config.yml:190]: 20 minutes was a bit excessive, but I added the setting when the tests were unpredictable, and I wasn't sure how long the longest run would take.

The build is more predictable, and it's rarely gone over 10 minutes.

**Pull request merged**